src/main/java/frc/robot/subsystems/ball/feeder/UltrasonicPoller.java

```diff
@@ -47,24 +48,27 @@ public void setTime(long time) {
     public void setDistance(double distance){
         this.distance = distance;
     }
-
-    // Consider renaming, probably want to generalize the name if we ever re-use it
-    public boolean checkUnloaded(){
+    
+    public boolean checkUnloaded() {
         return this.noBalls;
     }
 
     @Override
-    public void task(){
-        this.currentRange = this.ultrasonicSensor.getRangeMM();
+    public void start() {
+        this.firstRun = true;
+        super.start();
+    }
 
-        if(this.startTime == 0){
-            this.startTime = System.nanoTime(); // I know about a do while, but this isn't contained within a while loop 
-        }
-        
-        if((this.currentRange > this.distance) && (TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - this.startTime)  > this.time)) {
-            this.noBalls = true;
-        } else if(this.currentRange <= this.distance) {
+    @Override
+    public void task(){
+        if(firstRun) {
```

Comment:
You still haven't set `firstRun` to `false` within this branch, so `firstRun` will be `true` on every single run of the task.

Reply:
Ah missed these two. I did that but then messed up something in git and lost all my changes. Forgot to put those two back in

Reply:
Oh wait it was probably the merge conflict. But whatever